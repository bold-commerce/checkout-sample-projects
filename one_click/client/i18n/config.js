import i18n from 'i18next';
import enTranslation from './en/translation.json';
import { initReactI18next } from 'react-i18next';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: enTranslation
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false, 
    }
  });

export default i18n;