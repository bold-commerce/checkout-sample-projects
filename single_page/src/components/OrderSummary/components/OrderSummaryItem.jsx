import Price from '@boldcommerce/stacks-ui/lib/components/price/Price';
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const OrderSummaryItem = ({
  title,
  amount,
  lines,
}) => {
  return (
    <div className={classNames('OrderSummaryItem', lines ? 'OrderSummaryItem--removable' : '')}>
      <span className="OrderSummaryItem__Title">{title}</span>
      { lines ? lines : (
        amount > 0 && <span className="OrderSummaryItem__Amount">(<Price amount={amount}/>)</span>
      )}
    </div>
  );
};

OrderSummaryItem.propTypes = {
  title: PropTypes.string,
  amount: PropTypes.number,
  lines: PropTypes.node,
};

export default OrderSummaryItem;
