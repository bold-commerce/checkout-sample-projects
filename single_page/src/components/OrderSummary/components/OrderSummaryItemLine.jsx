import { Price } from '@boldcommerce/stacks-ui';
import React from 'react';
import RemoveIcon from './RemoveIcon';

const OrderSummaryItemLine = ({ description, amount, onRemove }) => (
  <div className="OrderSummaryItem__Description">
    { onRemove && (
      <button className="OrderSummaryItem__Action" type="button" onClick={onRemove}><RemoveIcon /></button>
    )}
    <span className="OrderSummaryItem__Label">{description}</span>
    <span className="OrderSummaryItem__Amount">(<Price amount={amount}/>)</span>
  </div>
);

export default OrderSummaryItemLine;
