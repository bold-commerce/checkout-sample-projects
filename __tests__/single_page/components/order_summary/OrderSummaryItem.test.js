import React from 'react';
import { render } from "@testing-library/react";
import OrderSummaryItem from "../../../../single_page/src/components/OrderSummary/components/OrderSummaryItem";
import OrderSummaryItemLine from "../../../../single_page/src/components/OrderSummary/components/OrderSummaryItemLine";
import '../../../../single_page/src/i18n/config';

describe('OrderSummaryLine', () => {
  test('renders OrderSummaryLine component', () => {
    const { asFragment } = render ( <OrderSummaryItem title={'Test Item'} /> );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders OrderSummaryLine component with amount', () => {
    const { asFragment } = render (
      <OrderSummaryItem
        title={'Test Item'}
        amount={1919}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders OrderSummaryLine component with lines', () => {
    const { asFragment } = render (
      <OrderSummaryItem
        title={'Test Item'}
        lines={
          <OrderSummaryItemLine
            description={'Test Card'}
            amount={1919}
            key={'lock'}
          />
        }
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
