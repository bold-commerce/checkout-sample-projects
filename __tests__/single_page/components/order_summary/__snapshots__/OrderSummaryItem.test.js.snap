// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderSummaryLine renders OrderSummaryLine component 1`] = `
<DocumentFragment>
  <div
    class="OrderSummaryItem"
  >
    <span
      class="OrderSummaryItem__Title"
    >
      Test Item
    </span>
  </div>
</DocumentFragment>
`;

exports[`OrderSummaryLine renders OrderSummaryLine component with amount 1`] = `
<DocumentFragment>
  <div
    class="OrderSummaryItem"
  >
    <span
      class="OrderSummaryItem__Title"
    >
      Test Item
    </span>
    <span
      class="OrderSummaryItem__Amount"
    >
      (
      <div
        class="stx-price stx-price--alight-center"
      >
        <p
          aria-label="Product price"
          class="stx-price__amount"
        >
          $19.19 CAD
        </p>
      </div>
      )
    </span>
  </div>
</DocumentFragment>
`;

exports[`OrderSummaryLine renders OrderSummaryLine component with lines 1`] = `
<DocumentFragment>
  <div
    class="OrderSummaryItem OrderSummaryItem--removable"
  >
    <span
      class="OrderSummaryItem__Title"
    >
      Test Item
    </span>
    <div
      class="OrderSummaryItem__Description"
    >
      <span
        class="OrderSummaryItem__Label"
      >
        Test Card
      </span>
      <span
        class="OrderSummaryItem__Amount"
      >
        (
        <div
          class="stx-price stx-price--alight-center"
        >
          <p
            aria-label="Product price"
            class="stx-price__amount"
          >
            $19.19 CAD
          </p>
        </div>
        )
      </span>
    </div>
  </div>
</DocumentFragment>
`;
